### PR TITLE
Subscriptions: Add the Subscribe Overlay toggle

### PR DESCRIPTION
**Pull request by pkuliga**
Related to https://github.com/Automattic/jetpack/issues/36208

## Proposed Changes

It adds the Subscribe Overlay toggle to Newsletter settings.

Screenshot 2024-05-24 at 10 36 25
<br><br>

This D149890-code is required to be merged first to make this change work.

## Testing Instructions

1. Go to Newsletter Settings
2. Make sure the toggle and "Preview and edit" link works as expected

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/site-settings/wrap-settings-form.jsx [hunk 1]
```diff
@@ -198,6 +198,12 @@ const wrapSettingsForm = ( getFormSettings ) => ( SettingsForm ) => {
 							path,
 						} );
 						break;
+					case 'jetpack_subscribe_overlay_enabled':
+						trackTracksEvent( 'calypso_settings_subscription_overlay_updated', {
+							value: fields.sm_enabled,
```
Comment (simison) [inline client/my-sites/site-settings/wrap-settings-form.jsx]: `sm_enabled` needs changing to `jetpack_subscribe_overlay_enabled`

Comment (pkuliga) [reply client/my-sites/site-settings/wrap-settings-form.jsx]: fixed, thanks!

**Review by simison — APPROVED** [review]
Tests well!

File: client/my-sites/site-settings/settings-newsletter/SubscribeOverlaySetting.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		<ToggleControl
+			checked={ !! value }
+			onChange={ handleToggle( SUBSCRIBE_OVERLAY_OPTION ) }
+			disabled={ disabled }
+			label={
+				<>
+					{ translate( 'Subscription overlay on homepage' ) }
+					{ subscribeOverlayEditorUrl && (
+						<>
+							{ '. ' }
```
Comment (simison) [inline client/my-sites/site-settings/settings-newsletter/SubscribeOverlaySetting.tsx:53]: Ideally this bit would be part of the string to be translated

**Pull request merged**